### PR TITLE
Document python.use_system_site_packages option

### LINKED ISSUE #4418: Document YAML python.use_system_site_packages option
A user raised (https://github.com/rtfd/readthedocs.org/issues/4362#issuecomment-404766625) that it was hard to find this option

```yaml
python:
   use_system_site_packages: true
```

I checked the docs and I didn't find anything about this: https://docs.readthedocs.io/en/latest/yaml-config.html

We need to extend our docs to mention this in the YAML config page.

### PR DESCRIPTION
**Pull request by stsewd**
Closes #4418

### CODE REVIEW

File: docs/yaml-config.rst [hunk 1]
```diff
@@ -154,6 +154,21 @@ the highest supported minor version will be selected.
...
        version: 3.5
 
+python.use_system_site_packages
+```````````````````````````````
+
+* Default: ``false``
+* Type: Boolean
+
+When true, it gives the virtual environment access to the global site-packages directory.
+Depending of the :ref:`yaml-config:build.image`,
```
Comment (agjohnson) [inline docs/yaml-config.rst]: Oops! Missed typo here. `Depending of` -> `Depending on`.

Comment (stsewd) [reply docs/yaml-config.rst]: I need to take a look at some english grammar rules p:

File: docs/yaml-config.rst [hunk 2]
```diff
@@ -154,6 +154,21 @@ the highest supported minor version will be selected.
...
 
+python.use_system_site_packages
+```````````````````````````````
+
+* Default: ``false``
+* Type: Boolean
+
+When true, it gives the virtual environment access to the global site-packages directory.
+Depending on the :ref:`yaml-config:build.image`,
+Read the Docs includes some libraries like scipy, numpy, etc.
```
Comment (humitos) [inline docs/yaml-config.rst]: Same as #4419 

If we don't want / can't document exactly what are those libraries, I think we should at least link to the repository that contains the `Dockerfile` and tell this to the user. I mean, how they can do to know exactly what are the pre-installed libs of the docker image they are using.

Comment (stsewd) [reply docs/yaml-config.rst]: Maybe is worth having a `build-images` docs? (or creating docs in the docker images repo)

Comment (humitos) [reply docs/yaml-config.rst]: I like the idea of having a page (inside the current docs) for the docker images that we use.

This page could explain what's the default image, explain or release process, what's installed on each of them, what's its base OS and how to check what specific version is installed on it.

Among other things 😁

Comment (stsewd) [reply docs/yaml-config.rst]: Well, I just discovered that we already have some docs about the packages installed, we should just improve/update them https://docs.readthedocs.io/en/latest/builds.html#packages-installed-in-the-build-environment

Comment (humitos) [reply docs/yaml-config.rst]: Good! So, in that case, I'd say that we should link that documentation page from here. Seems to be the most accurate to do and easy to follow.

Comment (humitos) [reply docs/yaml-config.rst]: We may need a different section for Python packages in those docs, though.

**Comment by humitos** [comment]
I'd say that this is ready to merge once we add the link to the build images.

File: docs/yaml-config.rst (hunk 2 above)
Comment (stsewd) [reply docs/yaml-config.rst]: Already did that haha, I thought you already saw the PR #4515

**Comment by stsewd** [comment]
Rebased and updated

File: docs/yaml-config.rst (hunk 1 above)
Comment (humitos) [reply docs/yaml-config.rst]: Please, fix this and we can merge, right?

Comment (stsewd) [reply docs/yaml-config.rst]: Ups, forgot to resolve this

**Pull request merged**